Mds/Collivery.php

```diff
@@ -98,850 +66,887 @@ protected function client()
+            }
+        }
+    }
+
+    /**
+     * Returns the status tracking detail of a given Waybill number.
+     * If the collivery is still active, the estimated time of delivery
+     * will be provided. If delivered, the time and receivers name (if availble)
+     * with returned.
+     *
+     * @param int $collivery_id Collivery ID
+     *
+     * @return array Collivery Status Information
+     * @throws Exception
+     */
+    public function getStatus($collivery_id)
+    {
+        try {
+            $result = $this->consumeAPI("https://api.collivery.co.za/v3/status_tracking/".$collivery_id, [], 'GET');
+        } catch (Exception $e) {
+            $this->catchException($e);
+
+            return false;
+        }
+        if (isset($result['data'])) {
+            if ($this->check_cache) {
+                $this->cache->put('collivery.status.'.$this->client_id.'.'.$collivery_id, $result['data'], 60 * 12);
+            }
+            return $result;
+        } else {
+            return $this->checkError($result);
+        }
+    }
+
+    /**
+     * Create a new Address and Contact.
+     *
+     * @param array $data Address and Contact Information
+     *
+     * @return array Address ID and Contact ID
+     * @throws Exception
+     */
+    public function addAddress(array $data)
+    {
+        $location_types = $this->make_key_value_array($this->getLocationTypes());
+        $towns = $this->make_key_value_array($this->getTowns());
+        $suburbs = $this->make_key_value_array($this->getSuburbs($data['town_id']));
+
+        if (!isset($data['location_type'])) {
+            $this->setError('missing_data', 'location_type not set.');
+        } elseif (!isset($location_types[$data['location_type']])) {
+            $this->setError('invalid_data', 'Invalid location_type.');
+        }
+
+        if (!isset($data['town_id'])) {
+            $this->setError('missing_data', 'town_id not set.');
+        } elseif (!isset($towns[$data['town_id']])) {
+            $this->setError('invalid_data', 'Invalid town_id.');
+        }
+
+        if (!isset($data['suburb_id'])) {
+            $this->setError('missing_data', 'suburb_id not set.');
+        } elseif (!isset($suburbs[$data['suburb_id']])) {
+            $this->setError('invalid_data', 'Invalid suburb_id.');
+        }
+
+        if (!isset($data['street'])) {
+            $this->setError('missing_data', 'street not set.');
+        }
+
+        if (!isset($data['contact']['full_name'])) {
+            $this->setError('missing_data', 'full_name not set.');
+        }
+
+        if (!isset($data['contact']['work_phone']) and !isset($data['contact']['cellphone'])) {
+            $this->setError('missing_data', 'Please supply ether a phone or cellphone number...2');
+        }
+
+        if (isset($data["custom_id"]) && is_integer($data["custom_id"])) {
+            $data["custom_id"] = $data["custom_id"] == 0 || $data["custom_id"] == "0" ? null : $data["custom_id"];
```

Comment:
Line 666 to 668  can be replaced with:
```php
$data["custom_id"] = $data["custom_id"] ? null : $data["custom_id"];
```
Being that `0`, `"0"`, `null` are all falsey

We also need to allow for custom ids that are not numeric